types & grammar/apA.md

```diff
@@ -1,227 +1,229 @@
 
-We'll briefly explore some of these concerns.
+Nós vamos explorar brevemente alguns destes casos.
 
-## Annex B (ECMAScript)
+## Anexo B (ECMAScript)
 
-It's a little known fact that the official name of the language is ECMAScript (referring to the ECMA standards body that manages it). What then is "JavaScript"? JavaScript is the common tradename of the language, of course, but more appropriately, JavaScript is basically the browser implementation of the spec.
+É um pouco conhecido o fato que o nome oficial da linguagem é ECMAScript (referindo-se ao conjunto de padrões ECMA que o gerencia). O que então é "JavaScript"? JavaScript é o nome comercial da linguagem, é claro, mas mais apropriadamente, JavaScript é basicamente a implementação desta especificação nos navegadores.
 
-The official ECMAScript specification includes "Annex B," which discusses specific deviations from the official spec for the purposes of JS compatibility in browsers.
+A especificação oficial ECMAScript inclui o "Anexo B", que discute derivações específicas da especificação oficial para os propósitos de compatibilidaded do JS nos navegadores.
 
-The proper way to consider these deviations is that they are only reliably present/valid if your code is running in a browser. If your code always runs in browsers, you won't see any observable difference. If not (like if it can run in node.js, Rhino, etc.), or you're not sure, tread carefully.
+O jeito apropriado de considerar essas derivações é que elas são confiavelmente válidas/presentes somente se seu código estiver rodando em um navegador. Se seu código sempre roda em navegadores, você não verá nenhuma diferença considerável. Caso contrário (como se pudesse ser executado em node.js, Rhino, etc.), ou você não tem certeza, vá com cuidado.
 
-The main compatibility differences:
+As pricipais diferenças de compatibilidade:
 
-* Octal number literals are allowed, such as `0123` (decimal `83`) in non-`strict mode`.
-* `window.escape(..)` and `window.unescape(..)` allow you to escape or unescape strings with `%`-delimited hexadecimal escape sequences. For example: `window.escape( "?foo=97%&bar=3%" )` produces `"%3Ffoo%3D97%25%26bar%3D3%25"`.
-* `String.prototype.substr` is quite similar to `String.prototype.substring`, except that instead of the second parameter being the ending index (noninclusive), the second parameter is the `length` (number of characters to include).
+* Números Octais são permitidos, como `0123` (decimal `83`) em non-`strict mode`.
+* `window.escape(..)` e `window.unescape(..)` permitem que você escape ou não strings com `%`-sequências de escape delimitadas. Por exemplo: `window.escape( "?foo=97%&bar=3%" )` se torna `"%3Ffoo%3D97%25%26bar%3D3%25"`.
+* `String.prototype.substr` é bem similiar à `String.prototype.substring`, exceto que ao invés do segundo parâmetro ser o último index (noninclusive), o segundo parâmetro é o `lenght` (números de caracteres para incluir).
```

Comment:
na parte _"the ending index (noninclusive)"_, o "noninclusive" se refere  que a ultima index não se inclui na resposta (não-inclusiva ou não-inclusa). Considere:

```js
var aline = 'aline'
aline.substring(0, 2)
// retorna "al". Cada letra é um indice (index) representado por um numero
// a-l-i-n-e
// 0-1-2-3-4
// O segundo parametro (2) é um delimitador e exclui ela mesma.
// Por isso nao temos o *i* como resposta.
```

e em contrapartida:

```js
var aline = 'aline'
aline.substr(0, 2)
// tambem retorna *al*

// O primeiro parametro é o inicio, entao 0 nao inicia nada.
// Note que nao estamos mais falando de indices aqui. Para substr temos esse caso:
// a-l-i-n-e
// 1-2-3-4-5

// No caso de *.substr*, o segundo parametro (2) se refere ao tamanho da string
// (5 -- aline tem 5 letras), mas diferente de arrays (que comecam com indice 0).
// a propriedade .length comeca do numero 1. Entao ao declarar (2), ele pega os dois indices
// apos o *0*, que sao 1 e 2 (A e L).
// Se quiser testar, tente mudar o *0* para *1*, tipo aline.substr(1, 2). Ele vai retornar *li*
// porque voce determinou que a funcao deveria comecar do indice 1 (letra A) e ter um limite
// de 2 caracteres. Como o *a* é _nao-inclusivo_, ele pega o *i* e o *l*.
```

acho que a gente pode mudar `(noninclusive)` para `nao-inclusivo`.